app/src/features/Navbar/Navbar.tsx

```diff
@@ -0,0 +1,149 @@
+  <MuiLink {...props} component={RouterLink} />
+);
+
+const useStyles = makeStyles((theme) => ({
+  separator: {
+    fontSize: theme.typography.h5.fontSize,
+    fill: theme.palette.text.secondary,
+  },
+  mainCrumb: {
+    fontWeight: 500,
+  },
+  breadCrumbLinks: {
+    fontSize: "1.2rem",
+    "&:hover": {
+      textDecoration: "none",
+      opacity: "0.6",
+    },
+  },
+  icons: {
+    height: 12,
+    marginBottom: "-2px",
+    display: "flex",
+    justifyContent: "center",
+    alignItems: "center",
+  },
+  breadcrumbsContainer: {
+    display: "flex",
+    padding: `${theme.spacing(4)}px ${theme.spacing(5)}px`,
+    minHeight: theme.mixins.breadcrumbBar.height,
+    alignItems: "center",
+  },
+  children: { marginLeft: "auto" },
+}));
+
+type NavbarProps = MuiBreadcrumbsProps & {
+  editMappingButton?: JSX.Element;
+};
+
+const Navbar = ({ editMappingButton, ...props }: NavbarProps): JSX.Element => {
+  const classes = useStyles();
+  const { t } = useTranslation();
+  const { sourceId, mappingId } = useParams<{
+    sourceId?: string;
+    mappingId?: string;
+  }>();
+  const location = useLocation();
+
+  const { data: source } = useApiSourcesRetrieveQuery(
+    {
+      id: sourceId ?? "",
+    },
+    { skip: !sourceId }
+  );
+  const { data: mapping } = useApiResourcesRetrieveQuery(
+    {
+      id: mappingId ?? "",
+    },
+    { skip: !mappingId }
+  );
+
+  const isEtl = location.pathname.endsWith("batches");
+  const isSource = sourceId && location.pathname.endsWith(sourceId);
+
+  return (
+    <div className={classes.breadcrumbsContainer}>
+      <Breadcrumbs
+        {...props}
+        separator={
+          <Icon
+            iconSize={14}
+            icon={IconNames.CHEVRON_RIGHT}
+            className={classes.icons}
+          />
+        }
+        classes={{ separator: classes.separator }}
+      >
+        <Link
+          variant="h5"
+          color="textSecondary"
+          to="/"
+          className={classes.breadCrumbLinks}
+        >
+          {t("sources")}
+        </Link>
+        {source && (
+          <Link
+            className={clsx(classes.breadCrumbLinks, {
+              [classes.mainCrumb]: source && !mapping && !isEtl,
+            })}
+            variant="h5"
+            color={!isSource ? "textSecondary" : "textPrimary"}
+            to={`/sources/${source.id}`}
+          >
+            {source.name}
+          </Link>
+        )}
+        {isEtl && (
+          <Link
+            className={clsx(classes.breadCrumbLinks, {
+              [classes.mainCrumb]: isEtl,
+            })}
+            variant="h5"
+            color={!isEtl ? "textSecondary" : "textPrimary"}
+            to={location.pathname}
+          >
+            {t("ETLDashboard")}
+          </Link>
+        )}
+        {source && mapping && (
+          <div>
+            <MappingSelectButton mapping={mapping} source={source} />
+            {editMappingButton}
+          </div>
+        )}
```

Comment:
I suggest you put the content of `editMappingButton` here, along with the `Menu` component and its logic

Reply:
I made a component for it (but I can put it inside the component Navbar as you said)
But it looked cleaner with a new component